src/codegen/query_compiler.cpp

```diff
@@ -41,7 +41,8 @@ std::unique_ptr<Query> QueryCompiler::Compile(
 }
 
 bool QueryCompiler::IsSupported(const planner::AbstractPlan &plan) {
-  return QueryCompiler::IsSupported(plan, nullptr);
+  QueryCompiler::IsSupported(plan, nullptr);
+  return false;
 }
```

Comment:
You need to modify `IsSupported` to check for the cases you want to catch, and return false for what isn't supported in codegen.